# R_tutorials
**Chen**

Some R tutorials: foundation and bioinformatics. 
Although most code have been edited, the reference of the origin code were presented before the tables of contents, only for education purpose.

Based on *Getting Started with R: An Introduction for Biologists, 2ed (Oxford, 2017)*
### Table of Contents: Foundation
| Chapter    | Title                                     | Imported?   |
| ---------- | ----------------------------------------- |:--------:| 
| Foundation | [Data management with dplyr](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Data%20management%20with%20dplyr.r)     |   |
|  | [Visualising data with ggplot2 I](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Visualising%20data%20with%20ggplot2%20I.r) |   |
|  | [Statistics I](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Statistics%20I.r)      |   |
|  | [Statistics II](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Statistics%20II.r)    |   |
|  | [Generalized linear models](https://github.com/liuchen37/R_tutorials/blob/main/Foundation:%20Generalized%20linear%20models)     |   |
|  | Visualising data with ggplot2 II          | Created  |
| Reference  | [Themes in ggplot2](https://github.com/liuchen37/R_tutorials/blob/main/Reference:%20Themes%20in%20ggplot2.r)        | Imported |
| Practice   | [Inhibitory effect of DEA and 6-MP on T47D](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20Inhibitory%20effect%20of%20DEA%20and%206-MP%20on%20T47D.r) | Created  |
|    | [HRas mutation rate](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20Counting%20freqency%20of%20an%20object%20in%20a%20table.r) |   |
| Data       | [Compensation](https://github.com/liuchen37/R_tutorials/blob/main/compensation.csv)                    | Imported |
|        | [Breast Cancer](https://github.com/liuchen37/R_tutorials/blob/main/BC.csv)      | Imported |
|        | [Daphniagrowth](https://github.com/liuchen37/R_tutorials/blob/main/Daphniagrowth.csv)            | Imported |
|        | [GardenOzone](https://github.com/liuchen37/R_tutorials/blob/main/GardenOzone.csv)                    | Imported |
|        | [Ladybirds_morph_colour](https://github.com/liuchen37/R_tutorials/blob/main/ladybirds_morph_colour.csv)           | Imported |
|        | [Plant.growth.rate](https://github.com/liuchen37/R_tutorials/blob/main/plant.growth.rate.csv)         | Imported |
|        | [Weight-height](https://github.com/liuchen37/R_tutorials/blob/main/weight-height.csv)                   | Imported |
|        | [Cow Growth](https://github.com/liuchen37/R_tutorials/blob/main/growth.csv)          | Imported |
|        | [Limpet](https://github.com/liuchen37/R_tutorials/blob/main/limpet.csv)    | Imported |

Based on *R Bioinformatics Cookbook (Packt, 2019)*
### Table of Contents: Bioinformatics
| Chapter    | Title     | Origin   |
| ---------- | ----------------------------------------- |:--------:| 
| Quantitative RNAseq | Estimating differential expression with edgeR |  |
|  | Estimating differential expression with DESeq2 |  |
|  | Power analysis with powersimR |  |
|  | Unannotated transcribed regions |  |
|  | High expression region with bumphunter |  |
|  | Differential peak analysis |  |
|  | Estimating batch effects with SVA |  |
|  | Allele-specific expression analysis with Alleliclmbalance |  |
|  | RNAseq data visulaization |  |
| Genetic variants and HTS data | SNPs and indels from sequencing data with VariantTools |  |
|  | Predicting ORFs in long reference sequences |  |
|  | Plotting with karyploteR |  |
|  | Variants selection and classifying with VariantAnnotation |  |
|  | Genomic regions of interest |  |
|  | GWAS |  |
|  | Estomating the copy number at a locus of interest |  |
| Genes, proteins, domains and motifs | DNA motifs with universalmotif |  |
|  | Protein domains with PFAM and bio3d |  |
|  | InterPro domains |  |
|  | Multiple alignments of genes and proteins |  |
|  | Aligning genomic length sequences with DECIPHER |  |
|  | Machine learning for protein noval feature predicton |  |
|  | 3D structure protein alighm,ent with bio3d |  |
| Phylogenetics | Reading and writing varied tree formats with ape and treeio |  |
|  | Tree visualization with ggtree |  |
|  | Quantifying differences with treespace |  |
|  | Subtrees |  |
|  | Alighnment visualization |  |
|  | Reconstructing trees using phangorn |  |
| Metagenomics |
| Metagenomics |
| Metagenomics |
| Metagenomics |
| Metagenomics |
| Metagenomics |
| Proteomics |
| Proteomics |
| Proteomics |
| Proteomics |
| Proteomics |
| Proteomics |
| Proteomics |
| Visualization for publication and web |
| Visualization for publication and web |
| Visualization for publication and web |
| Visualization for publication and web |
| Visualization for publication and web |
| Visualization for publication and web |
| Databases and remote data sources |
| Databases and remote data sources |
| Databases and remote data sources |
| Databases and remote data sources |
| Databases and remote data sources |
| Databases and remote data sources |
| Databases and remote data sources |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Statistics and machine learning |
| Tidyverse and BioConductor |
| Tidyverse and BioConductor |
| Tidyverse and BioConductor |
| Tidyverse and BioConductor |
| Tidyverse and BioConductor |
| Tidyverse and BioConductor |
| Building and packaging |
| Building and packaging |
| Building and packaging |
| Building and packaging |
| Building and packaging |
| Building and packaging |
| Building and packaging |
| Practice   | [GEOquery](https://github.com/liuchen37/R_tutorials/blob/main/Practice:%20GEOquery.r)               | Created  |

All data are in .csv format.

**Keep updating, keep learning...**
